iroh-bytes/src/provider.rs

```diff
@@ -34,6 +34,11 @@ pub use ticket::Ticket;
 /// Events emitted by the provider informing about the current status.
 #[derive(Debug, Clone)]
 pub enum Event {
+    /// A new collection has been added via an RPC call
```

Comment:
Can we make this more generic, so it always happens, not just on RPC call, like the other events?